test/multi-ns/connection_test.go

```diff
@@ -0,0 +1,155 @@
+package multi_ns_test
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"os"
+	"os/exec"
+	"testing"
+	"time"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+var (
+	clientset *kubernetes.Clientset
+	clientDep = &appsv1.Deployment{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "apps/v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "simple-client",
+			Namespace: "default",
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas: int32Ptr(1),
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					"networkservicemesh.io/app": "simple-client",
+				},
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						"networkservicemesh.io/app": "simple-client",
+					},
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name:            "alpine-img",
+							Image:           "alpine:latest",
+							ImagePullPolicy: corev1.PullIfNotPresent,
+							Command:         []string{"tail", "-f", "/dev/null"},
+						},
+					},
+				},
+			},
+		},
+	}
+)
+
+func TestSingleClientMultiNS(t *testing.T) {
+	for testName, c := range map[string]struct {
+		networkServices []string
+		annotations     map[string]string
+	}{
+		"Connect to one NSE": {
+			[]string{"foo"},
+			map[string]string{"ns.networkservicemesh.io": "foo?app=vl3-nse-foo"},
+		},
+		"Connect to two NSEs": {
+			[]string{"foo", "bar"},
+			map[string]string{"ns.networkservicemesh.io": "foo?app=vl3-nse-foo,bar?app=vl3-nse-bar"},
+		},
+	} {
+		t.Logf("Running test case: %s", testName)
+		var deployments []appsv1.Deployment
+		var ip_addr string
+		flag.StringVar(&ip_addr, "REMOTE_IP", "127.0.0.1", "Set ENV to REMOTE_IP")
+		flag.Parse()
+		//Install network service
+		for _, name := range c.networkServices {
+			fmt.Printf("Installing Network Service %s\n", name)
+			os.Setenv("REMOTE_IP", ip_addr)
+			os.Setenv("SERVICENAME", name)
+			cmd := exec.Command("../../scripts/vl3/vl3_interdomain.sh")
```

Comment:
Extract the path into a variable or put a flag for specifying the location of the script.